### ISSUE TITLE
Partition selection inconsistent with Java based Kafka tools 

### ISSUE THREAD

**Issue by ryanbamford** [fluent/fluent-plugin-kafka#397]
We have fluentd setup to send events into kafka 
```
<match ** >
    @type kafka2

    brokers "#{ENV['KAFKA_BROKER_HOST']}:#{ENV['KAFKA_BROKER_PORT']}"
    sasl_over_ssl true
    username "#{ENV['KAFKA_BROKER_USERNAME']}"
    password "#{ENV['KAFKA_BROKER_PASSWORD']}"
    scram_mechanism "#{ENV['KAFKA_SCRAM_MECHANISM']}"
    ssl_ca_cert "#{ENV['KAFKA_CA_CERT_LOCATION']}"

    # topic rooting data
    default_topic        undefined-catchall  
 
    <format>
      @type json
    </format>

    <buffer topic>
      @type file
      path /fluentd/var/kafka-buffer/
      chunk_limit_size 2m
      queued_chunks_limit_size 500
      flush_mode interval
      flush_interval 10s
      flush_thread_count 20
      flush_at_shutdown true
      retry_max_interval 120
     
    </buffer>
    compression_codec gzip

    max_send_limit_bytes 2000000

  </match>
```
and are sending a message with 
```
"message_key":"mk-123456"
```
the same message is then being sent from the standard kafka-console-producer with the same message key. 
the message is being sent into the same kafka topic that has been set up with 25 partitions. 

the result we are seeing is 

* fluentd message -> Partition:3-mk-123456
* kConP message -> Partition:5-mk-123456

the messages are sent to different partitions ??? 

This is breaking the processing logic in kafka streams ... not all messages are showing up in the same consumers when partitions are being separated 

there seems to be a missmatch in the way the keys are being hashed. 

gem 'fluent-plugin-kafka', '0.16.0'
gem 'ruby-kafka', '1.3.0'
kafka -> strimzi kafka version 2.7.0

**Comment by ryanbamford** [comment]
Think i can see the issue ... from the ruby kafka site 

Supported partitioning schemes
In order for semantic partitioning to work a partition_key must map to the same partition number every time. The general approach, and the one used by this library, is to hash the key and mod it by the number of partitions. There are many different algorithms that can be used to calculate a hash. By default crc32 is used. murmur2 is also supported for compatibility with Java based Kafka producers.

To use murmur2 hashing pass it as an argument to Partitioner. For example:
```
Kafka.new(partitioner: Kafka::Partitioner.new(hash_function: :murmur2))
```
the current plugin doesn't seem to support the setting of the underlying partitioner which would fix my issue

**Comment by kenhys** [comment]
This issue will be solved by #405, but it depends on the unreleased feature https://github.com/zendesk/ruby-kafka/blob/master/CHANGELOG.md#unreleased
